File: docs/examples/record_api_swift/Sources/RecordApiDocs/List.swift
import Foundation
import TrailBase

func list(client: Client) async throws -> ListResponse<Movie> {
    try await client
        .records("movies")
        .list(
            pagination: Pagination(limit: 3),
            order: ["rank"],
            filters: [
                // Multiple filters on same column: watch_time between 90 and 120 minutes
                .Filter(column: "watch_time", op: .GreaterThanOrEqual, value: "90"),
                .Filter(column: "watch_time", op: .LessThan, value: "120"),
                // Date range: movies released between 2020 and 2023
                .Filter(column: "release_date", op: .GreaterThanOrEqual, value: "2020-01-01"),
                .Filter(column: "release_date", op: .LessThanOrEqual, value: "2023-12-31"),
            ]
        )
}

File: docs/examples/record_api_go/list.go
package record_api_docs

import "github.com/trailbaseio/trailbase/client/go/trailbase"

func List(client trailbase.Client) (*trailbase.ListResponse[Movie], error) {
	api := trailbase.NewRecordApi[Movie](client, "movies")

	limit := uint64(3)

	return api.List(&trailbase.ListArguments{
		Pagination: trailbase.Pagination{
			Limit: &limit,
		},
		Order: []string{"rank"},
		Filters: []trailbase.Filter{
			// Multiple filters on same column: watch_time between 90 and 120 minutes
			trailbase.FilterColumn{Column: "watch_time", Op: trailbase.GreaterThanOrEqual, Value: "90"},
			trailbase.FilterColumn{Column: "watch_time", Op: trailbase.LessThan, Value: "120"},
			// Date range: movies released between 2020 and 2023
			trailbase.FilterColumn{Column: "release_date", Op: trailbase.GreaterThanOrEqual, Value: "2020-01-01"},
			trailbase.FilterColumn{Column: "release_date", Op: trailbase.LessThanOrEqual, Value: "2023-12-31"},
		},
	})
}
